server/src/main/java/io/seata/server/session/GlobalSession.java

```diff
@@ -249,8 +251,7 @@ public BranchSession getBranch(long branchId) {
      * @return the sorted branches
      */
     public ArrayList<BranchSession> getSortedBranches() {
-        ArrayList<BranchSession> sorted = new ArrayList();
-        sorted.addAll(branchSessions);
+        ArrayList<BranchSession> sorted = new ArrayList<>(branchSessions);
```

Comment:
please sort the HashSet

> It makes no guarantees as to the iteration order of the set; in particular, it does not guarantee that the order will remain constant over time.

Reply:
@leizhiyuan sorry, but I just replace the `ArrayList` of `lifecycleListeners`, and the `branchSessions` is still `ArrayList`:

![image](https://user-images.githubusercontent.com/15965696/58926042-4e3fa880-877c-11e9-948b-a2c5b7b62f95.png)

I just add a `final` modifier to prevent it being modified in the future because it's used as a lock in `synchronized` block:

![image](https://user-images.githubusercontent.com/15965696/58926099-7929fc80-877c-11e9-9376-fc967ccd65fd.png)

Reply:
And I'm sure that `new ArrayList<>(branchSessions);` will return a new list that keep the original order